handlers/mist_triggers.go

```diff
@@ -0,0 +1,207 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/livepeer/catalyst-api/clients"
+	"github.com/livepeer/catalyst-api/errors"
+)
+
+type MistCallbackHandlersCollection struct {
+	MistClient  MistAPIClient
+	StreamCache *StreamCache
+}
+
+// Trigger dispatches request to mapped method according to trigger name
+// Only single trigger callback is allowed on Mist.
+// All created streams and our handlers (segmenting, transcoding, et.) must share this endpoint.
+// If handler logic grows more complicated we may consider adding dispatch mechanism here.
+func (d *MistCallbackHandlersCollection) Trigger() httprouter.Handle {
+	return func(w http.ResponseWriter, req *http.Request, params httprouter.Params) {
+		t := &Trigger{w: w, req: req}
```

Comment:
Do we need this type `Trigger` at all? I think you can pass `w` and `req` directly. To me this type looks weird, because it has one additional field `streamName` which is sometimes filled, sometimes now. So, I'd suggest just keeping `w`, `req`, and `streamName` as separate variables.